idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceJavaStaticMethodWithTopLevelFunctionInspection.kt

```diff
@@ -0,0 +1,121 @@
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = callExpressionVisitor(fun(call) {
+        val callee = call.calleeExpression ?: return
+        val dotQualified = call.getTopmostParentOfType<KtDotQualifiedExpression>() ?: return
+        val calleeText = callee.text
+        val replacement = replacements
+            .firstOrNull { it.javaMethodShortName == calleeText }
+            ?.takeIf { call.getResolvedCall(call.analyze(BodyResolveMode.PARTIAL))?.isCalling(FqName(it.javaMethodFqName)) == true }
+            ?: return
+        if (replacement.toExtensionFunction && call.valueArguments.isEmpty()) return
+        holder.registerProblem(
+            dotQualified,
+            TextRange(0, callee.endOffset - dotQualified.startOffset),
+            "Should be replaced with '${replacement.kotlinFunctionShortName}()'",
+            ReplaceWithTopLevelFunction(replacement)
+        )
+    })
+
+    private class ReplaceWithTopLevelFunction(private val replacement: Replacement) : LocalQuickFix {
+        override fun getName() = "Replace with '${replacement.kotlinFunctionShortName}()'"
+
+        override fun getFamilyName() = name
+
+        override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+            val dotQualified = descriptor.psiElement as? KtDotQualifiedExpression ?: return
+            val call = dotQualified.callExpression ?: return
+            val file = dotQualified.containingKtFile
+            val psiFactory = KtPsiFactory(call)
+            val valueArguments = call.valueArguments
+            if (replacement.toExtensionFunction) {
+                val receiverText = valueArguments.first().text
+                val argumentsText = valueArguments.drop(1).joinToString(separator = ", ") { it.text }
+                dotQualified.replaced(psiFactory.createExpression("$receiverText.${replacement.kotlinFunctionShortName}($argumentsText)"))
+                file.resolveImportReference(FqName(replacement.kotlinFunctionFqName)).firstOrNull()?.also {
+                    ImportInsertHelper.getInstance(project).importDescriptor(file, it)
+                }
+            } else {
+                val argumentsText = valueArguments.joinToString(separator = ", ") { it.text }
+                val replaced = dotQualified.replaced(psiFactory.createExpression("${replacement.kotlinFunctionFqName}($argumentsText)"))
+                ShortenReferences.DEFAULT.process(replaced)
+            }
+        }
+    }
+
+    private data class Replacement(
+        val javaMethodFqName: String, 
+        val kotlinFunctionFqName: String,
+        val toExtensionFunction: Boolean = false
+    ) {
+        private fun String.shortName() = takeLastWhile { it != '.' }
+
+        val javaMethodShortName = javaMethodFqName.shortName()
+
+        val kotlinFunctionShortName = kotlinFunctionFqName.shortName()
+    }
+
+    companion object {
+        private val replacements = listOf(
```

Comment:
It is better to use `linkedSetOf`. You should override `equals` and `hashCode` for `Replacement` with `javaMethodFqName` as the key.

Reply:
Use `linkedMapOf` with `javaMethodFqName` as the key and `Replacement` as the value, also you can remove `javaMethodFqName` from `Replacement`.

Reply:
Or you can convert this list to a map.